### PR TITLE
Optimize build and release pipelines

### PR DESCRIPTION
**Pull request by dudizimber**
This PR splits the build pipeline into 2, one for building and testing, and another for releasing the image to Docker hub when a Release is created.

## Summary by CodeRabbit

- **New Features**
    - Enhanced the CI/CD workflow to trigger on pull request reviews and added steps for more robust build and testing processes.
    - Improved Docker support with updates to Dockerfiles for better image building and dependency management.
- **Chores**
    - Updated project and test environment setup in Docker for improved reliability and performance.

### CODE REVIEW

File: tests/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,25 @@
-FROM redis:latest as builder
+ARG BASE_IMAGE=falkordb/falkordb-compiler
 
-ENV DEPS "automake peg libtool autoconf python python-setuptools python-pip python-dev wget build-essential cmake m4 git valgrind"
+ARG TARGETPLATFORM=linux/amd64
+
+FROM --platform=$TARGETPLATFORM $BASE_IMAGE as compiler
+
+ENV DEPS "automake peg libtool autoconf python-is-python3 2to3 python3.10-venv python3-pip wget build-essential cmake m4 git valgrind gcc musl-dev python3-dev"
```
Comment (dudizimber) [inline tests/Dockerfile:7]: Check `musl-dev` dep

File: tests/fuzz/process.py [hunk 2]
```diff
@@ -24,6 +24,7 @@ def make_connection(module):
 
 def issue_queries(graph, timeout):
     os.environ["PATH"] = os.pathsep.join([ENV['PATH'], site.getuserbase() + '/bin'])
+    os.environ["PATH"] = os.pathsep.join([ENV['PATH'], '/venv/bin'])
```
Comment (dudizimber) [inline tests/fuzz/process.py:27]: Make it single line

File: tests/requirements.txt [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
 click
-behave~=1.2
-pathos~=0.2.8
+pathos
 FalkorDB
 neo4j
+git+https://github.com/renatahodovan/grammarinator.git@7dace5450b7136d641d065f649b406a22ef4d7f7
+wheel
+rmtest
+backports.csv
```
Comment (dudizimber) [inline tests/requirements.txt:10]: Check dep `backports.csv`

File: tests/requirements.txt [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+psutil
+redis-py-cluster
+redisgraph
```
Comment (dudizimber) [inline tests/requirements.txt:13]: Remove

**Review by dudizimber — COMMENTED** [review]
Review

**Review by AviAvni — APPROVED** [review]

**Pull request merged**